### PR TITLE
Calculate the intermediate output path in CompileContext.NeedsRebuilding

### LINKED ISSUE #1101: dotnet build throws Argument Null Exception for projects referencing projects with localized resx files
ProjectA contains Test.fr.resx
ProjectB references ProjectA

dotnet build produces the following stack trace:

```
System.ArgumentNullException: Value cannot be null.
Parameter name: path1
   at System.IO.Path.Combine(String path1, String path2)
   at Microsoft.DotNet.Tools.Compiler.CompilerUtil.<>c__DisplayClass5_0.<GetCultureResources>b__6(<>f__AnonymousType4`2 <>h__TransparentIdentifier1) in D:\work\katana\cli\src\Microsoft.DotNet.Tools.Compiler\CompilerUtil.cs:line 90
   at System.Linq.Enumerable.<>c__DisplayClass7_0`3.<CombineSelectors>b__0(TSource x)
   at System.Linq.Enumerable.<>c__DisplayClass7_0`3.<CombineSelectors>b__0(TSource x)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Microsoft.DotNet.Tools.Compiler.CompilerUtil.GetCultureResources(Project project, String outputPath) in D:\work\katana\cli\src\Microsoft.DotNet.Tools.Compiler\CompilerUtil.cs:line 85
   at Microsoft.DotNet.Tools.Build.CompileContext.AddCultureResources(ProjectContext project, String outputPath, CompilerIO compilerIO) in D:\work\katana\cli\src\Microsoft.DotNet.Tools.Builder\CompileContext.cs:line 487
   at Microsoft.DotNet.Tools.Build.CompileContext.GetCompileIO(ProjectContext project, String buildConfiguration, String outputPath, String intermediaryOutputPath, ProjectDependenciesFacade dependencies) in D:\work\katana\cli\src\Microsoft.DotNet.Tools.Builder\CompileContext.cs:line 433
   at Microsoft.DotNet.Tools.Build.CompileContext.NeedsRebuilding(ProjectContext project, ProjectDependenciesFacade dependencies, String buildOutputPath, String intermediateOutputPath) in D:\work\katana\cli\src\Microsoft.DotNet.Tools.Builder\CompileContext.cs:line 100
   at Microsoft.DotNet.Tools.Build.CompileContext.DependencyNeedsRebuilding(ProjectContext project, ProjectDependenciesFacade dependencies) in D:\work\katana\cli\src\Microsoft.DotNet.Tools.Builder\CompileContext.cs:line 90
   at Microsoft.DotNet.Tools.Build.CompileContext.Compile(Boolean incremental) in D:\work\katana\cli\src\Microsoft.DotNet.Tools.Builder\CompileContext.cs:line 62
```

I am using a locally built copy of dotnet-cli that is at `rel/1.0.0`

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #1101

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @davidfowl

**Comment by davidfowl** [comment]
:shipit:

File: src/Microsoft.DotNet.Tools.Builder/CompileContext.cs [hunk 1]
```diff
@@ -406,6 +406,7 @@ public CompilerIO(List<string> inputs, List<string> outputs)
             var compilerIO = new CompilerIO(new List<string>(), new List<string>());
             var calculator = project.GetOutputPathCalculator(outputPath);
             var binariesOutputPath = calculator.GetOutputDirectoryPath(buildConfiguration);
+            var intermediateOutputPath = calculator.GetIntermediateOutputDirectoryPath(buildConfiguration, intermediaryOutputPath);
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Tools.Builder/CompileContext.cs]: Should we change these variable names? intermediate and intermediary and pretty hard to distinguish. Maybe use just one and re-assign here?

Comment (pranavkm) [inline src/Microsoft.DotNet.Tools.Builder/CompileContext.cs]: Reassignment sounds good.

**Pull request closed without merging**